app/models/policy.rb

```diff
@@ -94,8 +94,23 @@ def destroy_orphaned_business_objective
     audit_bo_autoremove(removed_bos)
   end
 
+  def supported_os_minor_versions
+    profile_os_major_version = initial_profile.os_major_version
+    supported_benchmarks.map do |bm|
+      SupportedSsg.by_ssg_version[bm.version]
+                  .select { |ssg| ssg.os_major_version == profile_os_major_version }
+                  .map(&:os_minor_version)
+    end.flatten.uniq
+  end
+
   private
 
+  def supported_benchmarks
+    Xccdf::Benchmark.joins(:profiles).where(profiles: { parent_profile_id: nil,
+                                                        ref_id: initial_profile.ref_id,
+                                                        upstream: false })
+  end
```

Comment:
This is something we could extract into a scope, under the benchmarks, with profile as a parameter. Then it could be called above as:
```
Xccdf::Benchmark.including_profile(initial_profile)
```